content/blog/your-first-class-based-microsoft-dsc-v3-resource.md

```diff
@@ -0,0 +1,511 @@
+| `Set()`    | Applies desired state   | Nothing (void)                        | All           |
+| `Export()` | Exports configuration   | Array of resource instances           | V3 only       |
+
+## Why use class-based resources?
+
+Class-based DSC resources offer a modern, object-oriented approach to configuration management in PowerShell. Benefits include:
+
+- **Broad compatibility**: Works with both DSC v3 and legacy PSDSC engines
+- **Cleaner code**: Encapsulates logic in a structured, maintainable way
+- **Cross-platform support**: Runs on Windows, Linux, and macOS with proper implementation
+
+### Are class-based resources the future of DSC?
+
+While class-based resources are currently the recommended pattern for PowerShell DSC resource authoring, DSC v3 supports multiple languages and resource types. You can author resources in [Go][00], C#, Python and other languages. Class-based resources remain well-supported but may not be the only model in the future.
+
+For example, if your organization migrates from Windows-only infrastructure to a mix of Windows and Linux, or adopts newer versions of PowerShell, class-based resources help ensure your automation investments remain valid and portable across both the new Microsoft DSC v3 engine and the older PSDSC engine.
+
+In the next section, you will learn to set up a project structure with a class-based DSC resource named `DemoDscClass`.
+
+## Step 1: Create the project structure
+
+Create a new folder for your DSC resource module:
+
+  ```bash
+  # Create and navigate to project folder
+  mkdir DemoDscClass
+  cd DemoDscClass
+  ```
+
+Your project structure should look like this:
+
+  ```plaintext
+  DemoDscClass/
+  ├── DemoDscClass.psd1    # Module manifest
+  └── DemoDscClass.psm1    # Resource implementation
+  ```
+
+> [!IMPORTANT]
+> Your module manifest filename must match the project folder name. For example, if your folder is named `DemoDscClass`, your manifest must be `DemoDscClass.psd1`.
+
+## Step 2: Create the module manifest
+
+### Use PowerShell to generate the manifest
+
+  ```powershell
+  $manifestParams = @{
+      Path                  = '.\DemoDscClass.psd1'
+      RootModule            = 'DemoDscClass.psm1'
+      DscResourcesToExport  = 'DemoDscClass'
+      ModuleVersion         = '0.0.1'
+      CompatiblePSEditions  = @('Desktop', 'Core')
+      PowerShellVersion     = '5.1'
+  }
+
+  New-ModuleManifest @manifestParams
+  ```
+
+### Or create manually
+
+You can also create `DemoDscClass.psd1` using any text editor:
+
+  ```powershell
+  @{
+      RootModule = 'DemoDscClass.psm1'
+      ModuleVersion = '0.0.1'
+      CompatiblePSEditions = 'Desktop', 'Core'
+      GUID = '03ac4e95-e504-468d-add4-f099e2368239' # Change the GUID using New-Guid command
+      Author = 'username'
+      CompanyName = 'Unknown'
+      Copyright = '(c) developer. All rights reserved.'
+      PowerShellVersion = '5.1'
+      FunctionsToExport = '*'
+      CmdletsToExport = '*'
```

Comment:
Is it worth mentioning this problem with Azure https://github.com/dsccommunity/SqlServerDsc/pull/2110?

```suggestion

```

Reply:
`New-ModuleManifest @manifestParams` will generate that property so it has to be manually removed. Not sure setting it to '*' will resolve it in Azure, will ask Raimund. 🤔